public/app/core/navigation/patch/interceptLinkClicks.ts

```diff
@@ -14,6 +15,8 @@ export function interceptLinkClicks(e: MouseEvent) {
     const target = anchor.getAttribute('target');
 
     if (href && !target) {
+      const params = urlUtil.parseKeyValue(href.split('?')[1]);
```

Comment:
there's a locationSearchToObject method in LocationService. Maybe we can reuse it

Reply:
Done on https://github.com/grafana/grafana/pull/55837/commits/f1f6c6c29ea1a988b96fa04a1b4a57a0a0e97c75

Reply:
Reverted on https://github.com/grafana/grafana/pull/55837/commits/557a3c7b5770dc788d8fbcdb166a33a30c7780ea due to this function locationSearchToObject not dealing well with undefined values.